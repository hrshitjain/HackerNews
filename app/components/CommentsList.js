import React, { useState } from 'react';
import { View } from 'react-native';
import { fetchNewsItem } from '../services/APIService';
import Comment from './Comment';

const CommentsList = (props) => {
    const [replies, setReplies] = useState(null);

    const fetchReplies = (params) => {
        console.log('fetchReplies')
        const actions = params.kids.map(fetchNewsItem);
          var results = Promise.all(actions);
          results.then(data => {
            setReplies(data);
          }); 
    }
    const commentProps = {...props, fetchReplies: fetchReplies, children: replies ? replies.map(reply => Comment(reply)) : null };

    return(
        <View>
            {Comment(commentProps)}
        </View>
    );
}

export default CommentsList;