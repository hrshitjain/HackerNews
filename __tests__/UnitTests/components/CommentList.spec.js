import React from 'react';
import renderer from 'react-test-renderer';
import CommentsList from '../../../app/components/CommentsList';

describe('commentlist component render', () =>{
    const component = renderer.create(<CommentsList/>).toJSON();

    it('snapshot test', () =>{
        expect(component).toMatchSnapshot();
    });
    
});