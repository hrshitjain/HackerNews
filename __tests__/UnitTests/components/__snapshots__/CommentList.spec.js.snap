// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commentlist component render snapshot test 1`] = `
<View>
  <View
    style={
      Object {
        "borderColor": "red",
        "borderWidth": 2,
        "margin": 10,
        "marginTop": 0,
        "padding": 10,
      }
    }
  >
    <View>
      <Text />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "alignSelf": "flex-end",
              "fontWeight": "bold",
              "margin": 10,
            }
          }
        >
          Load replies...
        </Text>
      </View>
    </View>
  </View>
</View>
`;
